pygrametl/tables.py

```diff
@@ -1310,20 +1320,35 @@ def _after_insert(self, row, namemapping, newkeyvalue):
             tmp[self.key] = newkeyvalue
             self._after_getbykey(newkeyvalue, tmp)
 
-    def __performtype1updates(self, updates, lookupvalues, namemapping={}):
+    def __preparetype1updates(self, updates, lookupvalues, type2changes):
+        """ """
+        # Perform type 1 updates for the latest version
+        updateslatest = { att:value for (att, value) in updates.items()
+                          if not self.type1attsupdateall[att] }
+        if updateslatest and not type2changes:
+            self.__performtype1updates([lookupvalues[self.key]], updateslatest)
+
```

Comment:
Perhaps add a comment stating that when type2changes is True, the new version to be added already represents the change of the type1 attributes that only are updated in the newest version. Or does it get more confusing with such an explanation?

Reply:
Fixed.